smoke-test/src/test/java/org/opennms/smoketest/GrafanaEndpointPageIT.java

```diff
@@ -31,30 +32,65 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.stream.Collectors;
+import okhttp3.Credentials;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.MediaType;
+import okhttp3.Response;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
 
 import org.elasticsearch.common.Strings;
+import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 import org.opennms.netmgt.endpoints.grafana.api.GrafanaEndpoint;
 import org.opennms.smoketest.rest.GrafanaEndpointRestIT;
 import org.opennms.smoketest.ui.framework.Button;
 import org.opennms.smoketest.ui.framework.TextInput;
 import org.openqa.selenium.By;
+import org.openqa.selenium.TimeoutException;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.openqa.selenium.support.ui.WebDriverWait;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
 
 public class GrafanaEndpointPageIT extends UiPageTest  {
 
+
+    private static final Logger LOG = LoggerFactory.getLogger(GrafanaEndpointPageIT.class);
+
     private Page uiPage;
+    private static GenericContainer<?> grafanaContainer;
+    private static final String GRAFANA_URL = "http://admin:admin@localhost";
+    private static OkHttpClient client;
+    private static String token;
+    private static ObjectMapper objectMapper;
+    private static final String USERNAME = "admin";
+    private static final String PASSWORD = "admin";
+    private static final String VIEWER= "Viewer";
+    private static final String NAME="my-service-account";
 
     @Before
-    public void setUp() throws IOException, InterruptedException {
+    public void setUp() throws Exception {
         // Delete all endpoints
         sendDelete("rest/endpoints/grafana");
 
+
         uiPage = new Page(getBaseUrlInternal());
         uiPage.open();
+        grafanaContainer = new GenericContainer<>(DockerImageName.parse("grafana/grafana:8.5.26"))
```

Comment:
Move this to main  org.opennms.smoketest.containers along with other containers.

Also we can use latest grafana , setup user accounts and token, it should work with token. In my local testing, it worked for me.